SmartDeviceLink-iOS/SmartDeviceLink/SDLLocalization.m

```diff
@@ -0,0 +1,198 @@
+
+    dispatch_once(&token, ^{
+        // get the locale identifier
+        NSString *localeIdentifier = [[NSLocale currentLocale] localeIdentifier];
+        // extract language
+        NSString *language = ([localeIdentifier length] >= 2 ? [[localeIdentifier substringWithRange:NSMakeRange(0, 2)] lowercaseString] : nil);
+        // extract region
+        NSString *region   = ([localeIdentifier length] >= 5 ? [[localeIdentifier substringWithRange:NSMakeRange(3, 2)] uppercaseString] : nil);
+        
+        NSBundle *defaultBundle = nil;
+        NSLocale *defaultLocale = nil;
+        NSBundle *fallbackBundle = nil;
+        NSLocale *fallbackLocale = nil;
+        
+        // create default bundle and locale for language AND region only
+        if (language != nil && region != nil) {
+            // create a new locale identifier matching the bundle path for language AND region
+            localeIdentifier = [NSString stringWithFormat:@"%@-%@", [language lowercaseString], [language uppercaseString]];
+            // try to create a bundle for language AND region
+            defaultBundle = [NSBundle bundleWithPath:[[NSBundle mainBundle] pathForResource:localeIdentifier ofType:@"lproj"]];
+            // do we have a bundle for language AND region?
+            if (defaultBundle != nil) {
+                // we do. create a locale for language AND region
+                defaultLocale = [NSLocale localeWithLocaleIdentifier:localeIdentifier];
+            }
+        }
+        
+        // create fallback bundle and locale for language ONLY
+        if (language != nil) {
+            // create a new locale identifier matching the bundle path for langauge ONLY
+            localeIdentifier = [language lowercaseString];
+            // try to create a bundle for language ONLY
+            fallbackBundle = [NSBundle bundleWithPath:[[NSBundle mainBundle] pathForResource:localeIdentifier ofType:@"lproj"]];
+            // do we have a bundle for language ONLY?
+            if (fallbackBundle != nil) {
+                // we do. create a locale for language ONLY
+                fallbackLocale = [NSLocale localeWithLocaleIdentifier:localeIdentifier];
+            }
+        }
+        
+        // do we have any bundle object?
+        if (defaultBundle == nil && fallbackBundle == nil) {
+            // we don't. use system objects
+            fallbackBundle = [NSBundle mainBundle];
+            fallbackLocale = [NSLocale currentLocale];
+        }
+        
+        object = [[SDLLocalization alloc] init];
+        [object setLanguage:language];
+        [object setRegion:region];
+        [object setDefaultBundle:defaultBundle];
+        [object setDefaultLocale:defaultLocale];
+        [object setFallbackBundle:fallbackBundle];
+        [object setFallbackLocale:fallbackLocale];
+    });
+    
+    return object;
+}
+
++ (instancetype)localizationForLanguage:(NSString *)language forRegion:(NSString *)region {
```

Comment:
Region is necessary and used by existing apps to differentiate US,GB,AU english as well as FR,CA,CH french or PT,BR Portuguese (and many more). Supporting regions allows localization to be more flexible for the app developer. Otherwise developers need to give up existing region based localization (a no go in my opinion).

Signature will be changed to `localizationForLanguage: region:`